UE/Application/ISmsDb.hpp

```diff
@@ -1,15 +1,27 @@
 #pragma once
 
+#include <vector>
 namespace ue
 {
+    enum class SmsState
+    {
+        NotViewed,
+        Viewed
+    };
+
 class Sms;
 
 class ISmsDb
 {
 public:
+    using SmsMessages = std::vector<std::pair<Sms, SmsState>>;
+
     virtual ~ISmsDb() = default;
 
     virtual void addReceivedSms(const Sms& sms) = 0;
+    [[nodiscard]] virtual const SmsMessages& getSmsMessages() = 0;
+    virtual void updateSmsState(const unsigned int) = 0;
+    [[nodiscard]] virtual const Sms &getSms(const unsigned) = 0;
```

Comment:
redundant const when passing by copy

[ES.107: Don’t use unsigned for subscripts, prefer gsl::index](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-subscripts)
use ```std::vector<T>::size_type``` for now, we will ask whether we could use gsl

formatting

Reply:
actually it should be SmsMessages::size_type